awscli/clidocs.py

```diff
@@ -716,3 +730,62 @@ def _line_has_tag(self, line):
 
     def doc_subitems_start(self, help_command, **kwargs):
         pass
+
+
+class GlobalsRenderer:
+    """
+    Renderer used to generate global options for the
+    Synopsis and Global Options sections.
+    """
+
+    def __init__(self, driver):
+        self._driver = driver
+        self._global_options = None
+        self._global_synopsis = None
+
+    @property
+    def global_options(self):
+        if not self._global_options:
+            self._generate_global_options()
+        return self._global_options
+    
+    @property
+    def global_synopsis(self):
+        if not self._global_synopsis:
+            self._generate_global_synopsis()
+        return self._global_synopsis
+    
+    def _generate_global_options(self):
+        doc = ReSTDocument()
+        arg_table = self._driver._get_argument_table()
```

Comment:
In general, we should avoid accessing private methods (i.e. ones that start with underscore) as there is no compatibility contract on these. Instead, let's try to access the argument table from public interfaces. I think we may be able to use `self._driver.create_help_command().arg_table`?